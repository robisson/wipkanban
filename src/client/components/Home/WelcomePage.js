// @flow
import React from "react";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import CssBaseline from "@material-ui/core/CssBaseline";
import Paper from "@material-ui/core/Paper";
import GridContainer from "../GridContainer";
import Button from "@material-ui/core/Button";

const styles: Object = theme => ({
  layout: {
    width: "auto",
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit * 2,
    [theme.breakpoints.up(800 + theme.spacing.unit * 2 * 2)]: {
      width: "50%",
      marginLeft: "auto",
      marginRight: "auto"
    }
  },
  h1: {
    width: "100%"
  },
  paper: {
    marginBottom: theme.spacing.unit * 3,
    padding: theme.spacing.unit * 2,
    [theme.breakpoints.up(600 + theme.spacing.unit * 3 * 2)]: {
      marginBottom: theme.spacing.unit * 6,
      padding: theme.spacing.unit * 3
    }
  },
  buttons: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-end"
  },
  button: {
    marginLeft: theme.spacing.unit
  }
});

type Props = {
  classes: Object,
  onLogout: Function,
  onUpdateAccountUser: Function,
  user: Object
};

const WelcomePage = ({
  classes,
  onLogout,
  onUpdateAccountUser,
  user
}: Props) => {
  user.firstAccess = false;

  return (
    <React.Fragment>
      <CssBaseline />
      <GridContainer>
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <Typography
              className={classes.h1}
              component="h1"
              variant="h2"
              align="center"
              color="textPrimary"
              gutterBottom
            >
              Welcome to WipLean
            </Typography>
            <Typography variant="subtitle1" align="center">
              Before you start using WipLean, please complete your data below.
            </Typography>
            <Grid container spacing={24}>
              <Grid item lg={6}>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.name = target.value)}
                    defaultValue={user.name}
                    required
                    label="First name"
                    fullWidth
                    autoComplete="fname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.lastname = target.value)}
                    defaultValue={user.lastname}
                    required
                    label="Last name"
                    fullWidth
                    autoComplete="lname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.email = target.value)}
                    defaultValue={user.email}
                    required
                    label="Email"
                    fullWidth
                    autoComplete="billing address-level2"
                  />
                </Grid>
              </Grid>
              <Grid item lg={6}>
                <div
                  style={{
                    backgroundColor: "red",
                    width: "100%",
                    height: 250
                  }}
                />
                <input
                  type="file"
                  onChange={({ target }) => (user.image = target.files[0])}
                />
              </Grid>
              <Grid item className={classes.buttons}>
                <Button onClick={onLogout}>Logout</Button>
                <Button
                  onClick={() => onUpdateAccountUser(user)}
                  variant="contained"
                  color="primary"
                  className={classes.button}
                >
                  Update and next
                </Button>
              </Grid>
            </Grid>
          </Paper>
        </main>
      </GridContainer>
    </React.Fragment>
  );
};

export default withStyles(styles)(WelcomePage);
