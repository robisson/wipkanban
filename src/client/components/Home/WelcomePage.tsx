import * as React from "react";
import Typography from "@material-ui/core/Typography";
import {
  withStyles,
  StyleRulesCallback,
  Theme
} from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import CssBaseline from "@material-ui/core/CssBaseline";
import Paper from "@material-ui/core/Paper";
import GridContainer from "../GridContainer";
import Button from "@material-ui/core/Button";
import grey from "@material-ui/core/colors/grey";

const styles: StyleRulesCallback = (theme: Theme) => ({
  layout: {
    width: "auto",
    marginLeft: theme.spacing.unit * 2,
    marginRight: theme.spacing.unit * 2,
    [theme.breakpoints.up(800 + theme.spacing.unit * 2 * 2)]: {
      width: "50%",
      marginLeft: "auto",
      marginRight: "auto"
    }
  },
  h1: {
    width: "100%"
  },
  paper: {
    marginBottom: theme.spacing.unit * 3,
    padding: theme.spacing.unit * 2,
    [theme.breakpoints.up(600 + theme.spacing.unit * 3 * 2)]: {
      marginBottom: theme.spacing.unit * 6,
      padding: theme.spacing.unit * 3
    }
  },
  buttons: {
    width: "100%",
    display: "flex",
    justifyContent: "flex-end"
  },
  button: {
    marginLeft: theme.spacing.unit
  },
  photo: {
    backgroundColor: grey[300],
    width: 150,
    height: 200,
    margin: [`${theme.spacing.unit * 2}px`, "auto"].join(" ")
  }
});

interface IWelcomePageProps {
  classes: any;
  onLogout: any;
  onUpdateAccountUser: Function;
  user: any;
}

const WelcomePage: React.StatelessComponent<any> = ({
  classes,
  onLogout,
  onUpdateAccountUser,
  user
}: IWelcomePageProps) => {
  user.firstAccess = false;

  return (
    <React.Fragment>
      <CssBaseline />
      <GridContainer>
        <main className={classes.layout}>
          <Paper className={classes.paper}>
            <Typography
              className={classes.h1}
              component="h1"
              variant="h2"
              align="center"
              color="textPrimary"
              gutterBottom
            >
              Welcome to WipLean
            </Typography>
            <Typography variant="subtitle1" align="center">
              Before you start using WipLean, please complete your data below.
            </Typography>
            <Grid container spacing={24}>
              <Grid item lg={6}>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.name = target.value)}
                    defaultValue={user.name}
                    required
                    label="First name"
                    fullWidth
                    autoComplete="fname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.lastname = target.value)}
                    defaultValue={user.lastname}
                    required
                    label="Last name"
                    fullWidth
                    autoComplete="lname"
                  />
                </Grid>
                <Grid item>
                  <TextField
                    onChange={({ target }) => (user.email = target.value)}
                    defaultValue={user.email}
                    required
                    label="Email"
                    fullWidth
                    autoComplete="billing address-level2"
                  />
                </Grid>
              </Grid>
              <Grid item lg={6}>
                <div className={classes.photo} id="divGreyPhoto">
                  <img id="imagePreview" width="100%" />
                </div>
                <input
                  type="file"
                  id="contained-button-file"
                  onChange={({ target }: any) => {
                    user.image = target.files[0];

                    let reader = new FileReader();

                    reader.onload = function(e: any) {
                      // get loaded data and render thumbnail.
                      const image: HTMLElement = document.getElementById(
                        "imagePreview"
                      ) as HTMLElement;

                      if (!!image) {
                        (image as HTMLImageElement).src = e.target.result;
                      }
                    };

                    // read the image file as a data URL.
                    reader.readAsDataURL(target.files[0]);
                  }}
                  accept="image/*"
                  style={{ display: "none" }}
                />
                <div style={{ textAlign: "center" }}>
                  <label htmlFor="contained-button-file">
                    <Button variant="contained" component="span">
                      Choose image
                    </Button>
                  </label>
                </div>
              </Grid>
              <Grid item className={classes.buttons}>
                <Button onClick={onLogout}>Logout</Button>
                <Button
                  onClick={() => onUpdateAccountUser(user)}
                  variant="contained"
                  color="primary"
                  className={classes.button}
                >
                  Update and next
                </Button>
              </Grid>
            </Grid>
          </Paper>
        </main>
      </GridContainer>
    </React.Fragment>
  );
};

export default withStyles(styles)(WelcomePage);
