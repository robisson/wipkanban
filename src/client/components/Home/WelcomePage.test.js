import React from "react";
import { createShallow } from "@material-ui/core/test-utils";
import WelcomePage from "./WelcomePage";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import "../../setupTest";
import toJson from "enzyme-to-json";
import Typography from "@material-ui/core/Typography";
import Grid from "@material-ui/core/Grid";

describe("<WelcomePage /> component", () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it("should render component correctly", () => {
    let wrapper = shallow(<WelcomePage />);

    expect(wrapper.type()).toEqual(React.Fragment);
    expect(wrapper.find(Typography).length).toEqual(2);
    expect(wrapper.find(Grid).length).toEqual(9);
    expect(wrapper.find(TextField).length).toEqual(6);
    expect(wrapper.find(Button).length).toEqual(2);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should to call function logout when clicked at Button", () => {
    let onLogout = jest.fn();
    let wrapper = shallow(<WelcomePage onLogout={onLogout} />);

    wrapper
      .find(Button)
      .at(0)
      .simulate("click");
    expect(onLogout.mock.calls.length).toEqual(1);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
