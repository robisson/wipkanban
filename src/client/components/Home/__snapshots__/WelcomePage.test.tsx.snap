// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomePage /> component should render component correctly 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(GridContainer)>
    <main
      className="WelcomePage-layout-1"
    >
      <WithStyles(Paper)
        className="WelcomePage-paper-3"
      >
        <WithStyles(Typography)
          align="center"
          className="WelcomePage-h1-2"
          color="textPrimary"
          component="h1"
          gutterBottom={true}
          variant="h2"
        >
          Welcome to WipLean
        </WithStyles(Typography)>
        <WithStyles(Typography)
          align="center"
          variant="subtitle1"
        >
          Before you start using WipLean, please complete your data below.
        </WithStyles(Typography)>
        <WithStyles(Grid)
          container={true}
          spacing={24}
        >
          <WithStyles(Grid)
            item={true}
            lg={6}
          >
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="fname"
                fullWidth={true}
                label="First name"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="lname"
                fullWidth={true}
                label="Last name"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="billing address-level2"
                defaultValue="example@test.com"
                fullWidth={true}
                label="Email"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={6}
          >
            <div
              className="WelcomePage-photo-6"
              id="divGreyPhoto"
            >
              <img
                id="imagePreview"
                width="100%"
              />
            </div>
            <input
              accept="image/*"
              id="contained-button-file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <label
                htmlFor="contained-button-file"
              >
                <WithStyles(Button)
                  component="span"
                  variant="contained"
                >
                  Choose image
                </WithStyles(Button)>
              </label>
            </div>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            className="WelcomePage-buttons-4"
            item={true}
          >
            <WithStyles(Button)
              onClick={[MockFunction]}
            >
              Logout
            </WithStyles(Button)>
            <WithStyles(Button)
              className="WelcomePage-button-5"
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Update and next
            </WithStyles(Button)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Paper)>
    </main>
  </WithStyles(GridContainer)>
</Fragment>
`;

exports[`<WelcomePage /> component should to call function logout when clicked at Button 1`] = `
<Fragment>
  <WithStyles(CssBaseline) />
  <WithStyles(GridContainer)>
    <main
      className="WelcomePage-layout-1"
    >
      <WithStyles(Paper)
        className="WelcomePage-paper-3"
      >
        <WithStyles(Typography)
          align="center"
          className="WelcomePage-h1-2"
          color="textPrimary"
          component="h1"
          gutterBottom={true}
          variant="h2"
        >
          Welcome to WipLean
        </WithStyles(Typography)>
        <WithStyles(Typography)
          align="center"
          variant="subtitle1"
        >
          Before you start using WipLean, please complete your data below.
        </WithStyles(Typography)>
        <WithStyles(Grid)
          container={true}
          spacing={24}
        >
          <WithStyles(Grid)
            item={true}
            lg={6}
          >
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="fname"
                fullWidth={true}
                label="First name"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="lname"
                fullWidth={true}
                label="Last name"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
            <WithStyles(Grid)
              item={true}
            >
              <TextField
                autoComplete="billing address-level2"
                defaultValue="example@test.com"
                fullWidth={true}
                label="Email"
                onChange={[Function]}
                required={true}
                select={false}
                variant="standard"
              />
            </WithStyles(Grid)>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            item={true}
            lg={6}
          >
            <div
              className="WelcomePage-photo-6"
              id="divGreyPhoto"
            >
              <img
                id="imagePreview"
                width="100%"
              />
            </div>
            <input
              accept="image/*"
              id="contained-button-file"
              onChange={[Function]}
              style={
                Object {
                  "display": "none",
                }
              }
              type="file"
            />
            <div
              style={
                Object {
                  "textAlign": "center",
                }
              }
            >
              <label
                htmlFor="contained-button-file"
              >
                <WithStyles(Button)
                  component="span"
                  variant="contained"
                >
                  Choose image
                </WithStyles(Button)>
              </label>
            </div>
          </WithStyles(Grid)>
          <WithStyles(Grid)
            className="WelcomePage-buttons-4"
            item={true}
          >
            <WithStyles(Button)
              onClick={
                [MockFunction] {
                  "calls": Array [
                    Array [],
                  ],
                  "results": Array [
                    Object {
                      "type": "return",
                      "value": undefined,
                    },
                  ],
                }
              }
            >
              Logout
            </WithStyles(Button)>
            <WithStyles(Button)
              className="WelcomePage-button-5"
              color="primary"
              onClick={[Function]}
              variant="contained"
            >
              Update and next
            </WithStyles(Button)>
          </WithStyles(Grid)>
        </WithStyles(Grid)>
      </WithStyles(Paper)>
    </main>
  </WithStyles(GridContainer)>
</Fragment>
`;
